litestar/middleware/_internal/exceptions/middleware.py

```diff
@@ -202,7 +202,9 @@ async def handle_request_exception(
         exception_handler = get_exception_handler(exception_handlers, exc) or self.default_http_exception_handler
         request: Request[Any, Any, Any] = litestar_app.request_class(scope=scope, receive=receive, send=send)
         response = exception_handler(request, exc)
-        await response.to_asgi_response(app=None, request=request)(scope=scope, receive=receive, send=send)
+        await response.to_asgi_response(app=None, request=request, type_encoders=litestar_app.type_encoders)(
```

Comment:
```suggestion
        await response.to_asgi_response(app=None, request=request, type_encoders=litestar_app.type_encoders)(
```

This should try to retrieve the type encoders from the route handler, and only if one isn't found (e.g. during a `404`), fall back to the app's type encoders

Reply:
I don't see any changes

Reply:
> This should try to retrieve the type encoders from the route handler, and only if one isn't found (e.g. during a 404), fall back to the app's type encoders

Isn't that the default behaviour anyway?
https://github.com/litestar-org/litestar/blob/3de1b1dabd53b85e17eae351e4e73eddf71ce284/litestar/response/base.py#L262

Reply:
> I don't see any changes

Sorry, this wasn't meant to be a code suggestion, just a comment.

> Isn't that the default behaviour anyway?

No. The response isn't tied to the app/route handler's type encoders. You need to call `handler.resolve_type_encoders()` to do that.

Reply:
Ah, gotcha, there three levels of type encoders - app, handler and response and I've missed the handler

Reply:
> Ah, gotcha, there three levels of type encoders - app, handler and response and I've missed the handler

Actually, there's an unknown number of type encoders, since type encoders are supported on all layers (app, routers, controllers, handlers). The `handlers.resolve_type_encoders()` will merge those and give you the result back. In the happy path, this is not an issue since we'll always have access to a handler when creating the response, so we can just pass those in, but during exception handling, we have to do a bit of legwork :)

Reply:
yeah, I had a peek in that method. Thanks for the explanation!